### PR TITLE
VSI credentials followups

### PR DESCRIPTION
**Pull request by nyalldawson**
A couple of follow up fixes and cleanups for https://github.com/qgis/QGIS/pull/57826

### CODE REVIEW

File: src/core/qgsgdalutils.cpp [hunk 1]
```diff
@@ -965,21 +965,50 @@ bool QgsGdalUtils::isVsiArchiveFileExtension( const QString &extension )
...
+  if ( vsiPrefix == QLatin1String( "vsizip" ) ||
+       vsiPrefix == QLatin1String( "vsigzip" ) ||
+       vsiPrefix == QLatin1String( "vsitar" ) ||
+       vsiPrefix == QLatin1String( "vsi7z" ) ||
+       vsiPrefix == QLatin1String( "vsirar" ) )
+    return Qgis::VsiHandlerType::Archive;
+
+  else if ( vsiPrefix == QLatin1String( "vsicurl" ) ||
+            vsiPrefix == QLatin1String( "vsicurl_streaming" ) )
+    return Qgis::VsiHandlerType::Archive;
```
Comment (nyalldawson) [inline src/core/qgsgdalutils.cpp:991]:
```suggestion
    return Qgis::VsiHandlerType::Network;
```

File: tests/src/core/testqgsgdalprovider.cpp [hunk 2]
```diff
@@ -919,22 +920,57 @@ void TestQgsGdalProvider::testVsiCredentialOptions()
...
+  QVERIFY( subLayers.isEmpty() );
+
+  // confirm that GDAL VSI configuration options are set
+  QString noSign( VSIGetPathSpecificOption( "/vsis3/gdalsublayerstestbucket", "AWS_NO_SIGN_REQUEST", nullptr ) );
+  QCOMPARE( noSign, QStringLiteral( "YES" ) );
+  QString region( VSIGetPathSpecificOption( "/vsis3/gdalsublayerstestbucket", "AWS_REGION", nullptr ) );
+  QCOMPARE( region, QStringLiteral( "eu-central-3" ) );
+
+  subLayers = gdalMetadata->querySublayers( QStringLiteral( "/vsis3/gdalsublayerstestbucket/test.tif|credential:AWS_NO_SIGN_REQUEST=YES|credential:AWS_REGION=eu-central-3|credential:AWS_S3_ENDPOINT=localhost" ), Qgis::SublayerQueryFlag::FastScan );
+  // ideally we'd test with a real dataset here!
```
Comment (rouault) [inline tests/src/core/testqgsgdalprovider.cpp:968]: if the HDF5 driver is available:
```
gdalinfo HDF5:"/vsis3/noaa-goes16/ABI-L1b-RadC/2022/001/00/OR_ABI-L1b-RadC-M6C01_G16_s20220010001173_e20220010003546_c20220010003587.nc"://Rad --config AWS_NO_SIGN_REQUEST YES
```

Comment (nyalldawson) [reply tests/src/core/testqgsgdalprovider.cpp:968]: @rouault do you think that's safe to rely on in tests? I really don't want another flaky test 😂

Comment (rouault) [reply tests/src/core/testqgsgdalprovider.cpp:968]: That's the one we use in the GDAL /vsis3/ test. I haven't seen it flaky, but can't promise on the long term...

An alternative is also ``gdalinfo /vsis3/cdn.proj.org/us_nga_egm96_15.tif --config AWS_NO_SIGN_REQUEST YES`` . Should stay there while AWS is fine hosting us.

Comment (nyalldawson) [reply tests/src/core/testqgsgdalprovider.cpp:968]: Thanks! Do you know of any similar vector sources?

Comment (rouault) [reply tests/src/core/testqgsgdalprovider.cpp:968]: ``ogrinfo /vsis3/cdn.proj.org/files.geojson --config AWS_NO_SIGN_REQUEST YES``

Comment (nyalldawson) [reply tests/src/core/testqgsgdalprovider.cpp:968]: Thanks, done!

**Comment by nyalldawson** [comment]
@rouault does this look ok to you now?

**Review by rouault — APPROVED** [review]

**Comment by nyalldawson** [comment]
Thanks @rouault !

**Pull request merged**